.github/workflows/rust.yml

```diff
@@ -157,3 +157,21 @@ jobs:
 
       - name: Build
         run: cargo xtask build
+
+  # Run the build using an older nightly (specified in
+  # ./old_nightly_toolchain.toml). This serves to check that we don't
+  # accidentally start relying on a new feature without intending to.
+  # Having a test for this makes it easier for us to be intentional
+  # about making changes that require a newer version.
+  old_nightly_build:
```

Comment:
I think it'd be cool if this is called "build_msrv" or so. Also, I think we can drop the toolchain file. Just use `cargo +$MSRV xtask build` where `MSRV=nightly-2022-04-18`

Reply:
Updated the name as suggested.

I tried removing the extra toolchain file, but I forgot that then I'd need to manually call `rustup install ...` and `rustup component add rust-src --toolchain ...` in the job, which I think doesn't end up being any better than having a toolchain file. So I reverted those changes and kept the extra file.